forge/ee/db/controllers/Subscription.js

```diff
@@ -19,5 +19,12 @@ module.exports = {
             subscription.destroy()
         }
         return null
+    },
+
+    userEligibleForFreeTrial: async function (app, user, newTeamAlreadyCreated = false) {
+        const teams = await app.db.models.Team.forUser(user)
```

Comment:
@knolleary @hardillb This feels like a sub-optimal way to check the number of teams a user is a member of, but I couldn't find nicer way to do the count given it's via the TeamMembership table, any ideas?

Reply:
I would add a `getTeamCount` instance finder on the `User` model, which in turn calls `TeamMember.countUserTeams` static finder on the `TeamMember` model.

Reply:
... which now that I look at the changes made since this comment, I see you've done a slight variation of that which is just as good.